pkg/collector/corechecks/snmp/internal/devicecheck/devicecheck.go

```diff
@@ -152,15 +160,22 @@ func (d *DeviceCheck) Run(collectionTime time.Time) error {
 	var pingStatus metadata.DeviceStatus
 
 	deviceReachable, dynamicTags, values, checkErr := d.getValuesAndTags()
+	cachedTags := d.readCache()
```

Comment:
Instead of reading the tags from disk when the check is run should we read them when the check is created in NewDeviceCheck()?

Reply:
yes good idea ! https://github.com/DataDog/datadog-agent/pull/25534/commits/4174011a910ae2b72866da6e965118b820dd11b3 
I did a mix between the old solution with the `savedDynamicTags` variable for better performance